From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Nikola Milosavljevic <nikolam@microsoft.com>
Date: Thu, 6 Apr 2023 19:53:04 +0000
Subject: [PATCH] Only pack the shipping/non-shipping nupkgs for roslyn

Avoid packing release, pre-release stable nupkgs, and symbol nupkg.

Backport: https://github.com/dotnet/roslyn/pull/67679
---
 eng/SourceBuild.props | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/eng/SourceBuild.props b/eng/SourceBuild.props
index d5d41e7167e..a8b55f352ad 100644
--- a/eng/SourceBuild.props
+++ b/eng/SourceBuild.props
@@ -4,8 +4,21 @@
     <GitHubRepositoryName>roslyn</GitHubRepositoryName>
     <SourceBuildManagedOnly>true</SourceBuildManagedOnly>
     <SourceBuildTrimNetFrameworkTargets>true</SourceBuildTrimNetFrameworkTargets>
+    <!-- Roslyn produces stable release branded and stable pre-release branded artifacts in addition to the normal non-stable artifacts.
+         Only the non-stable artifacts should flow downstream in source build -->
+    <EnableDefaultSourceBuildIntermediateItems>false</EnableDefaultSourceBuildIntermediateItems>
   </PropertyGroup>
 
+  <Target Name="GetCustomIntermediateNupkgContents" BeforeTargets="GetCategorizedIntermediateNupkgContents">
+    <ItemGroup>
+      <IntermediateNupkgArtifactFile Include="$(CurrentRepoSourceBuildArtifactsPackagesDir)Shipping\**\*.nupkg" />
+      <IntermediateNupkgArtifactFile Include="$(CurrentRepoSourceBuildArtifactsPackagesDir)NonShipping\**\*.nupkg" />
+      <!-- Don't pack any symbol packages: not needed for downstream source-build CI.
+          Roslyn's symbol packages come in .Symbols.<version>.nupkg instead of the standard format. -->
+      <IntermediateNupkgArtifactFile Remove="$(CurrentRepoSourceBuildArtifactsPackagesDir)**\*.Symbols.*.nupkg" />
+    </ItemGroup>
+  </Target>
+
   <!--
     The build script passes in the full path of the sln to build.  This must be overridden in order to build
     the cloned source in the inner build.
